client/src/sections/courses/LearningSection/AddLearnerModal.tsx

```diff
@@ -1,125 +1,184 @@
 
-interface AddLearnerModalProps {
-  learners: Learner[];
-}
-
-const AddLearnerModal: React.FC<AddLearnerModalProps> = ({ learners }: AddLearnerModalProps): JSX.Element => {
-  const [currentPage, setCurrentPage] = useState(1);
+const AddLearnerModal: React.FC = (): JSX.Element => {
+  const router = useRouter();
+  const params = router.query;
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const [selectedEnrollees, setSelectedEnrollees] = useState<Trainee[]>([]);
+  const courseID = params.id;
+  const dispatch = useAppDispatch();
 
-  const handleAddLearners = (): void => {
-    alert('Add Learners');
-  };
+  const { isEnrollee } = useSelector((state: RootState) => state.courseLearner);
 
-  const handleModalClose = (): void => {
-    setIsModalOpen(false);
+  const [mutateAsync] = useEnrollLearnerMutation();
+
+  const { data: trainee, refetch } = useGetLearnerQuery({
+    courseId: courseID,
+    isEnrolled: isEnrollee,
+    pageNumber: page,
+    searchQuery: search,
+  });
+  const enrollees = trainee?.learners.data;
+
+  const handleAddMaterialModal = (): void => {
+    setSelectedIds([]);
+    setSearch('');
+    setPage(1);
+    setIsModalOpen(!isModalOpen);
+    dispatch(toggleLearner());
+    if (!isEnrollee) {
+      void refetch();
+    }
   };
 
   const handleSearch = (search: string): void => {
-    alert(`You searched ${search}`);
+    setSearch(search);
+    setPage(1);
   };
 
-  const learnersPerPage = 8;
-  const totalPages = Math.ceil(learners.length / learnersPerPage);
+  const handleChangePageEvent = (page: number): void => {
+    setPage(page);
+  };
 
-  const lastIndex = currentPage * learnersPerPage;
-  const firstIndex = lastIndex - learnersPerPage;
-  const currentLearners = learners.slice(firstIndex, lastIndex);
+  const handleCheckboxChange = (id: number): void => {
+    if (selectedIds.includes(id)) {
+      setSelectedIds(selectedIds.filter((selectedId) => selectedId !== id));
+      setSelectedEnrollees(selectedEnrollees.filter((enrollee) => enrollee.trainee_id !== id));
+    } else {
+      setSelectedIds([...selectedIds, id]);
+      const selectedEnrollee = enrollees.find((enrollee: Trainee) => enrollee.trainee_id === id);
+      if (selectedEnrollee) {
+        setSelectedEnrollees([...selectedEnrollees, selectedEnrollee]);
+      }
+    }
+  };
 
-  const handleChangePageEvent = (page: number): void => {
-    setCurrentPage(page);
+  const handleAddLearners = async (): Promise<void> => {
+    const postData = {
+      trainee: selectedIds.join(','),
+    };
+    try {
+      await mutateAsync({
+        courseId: courseID,
+        postData,
+      });
+      dispatch(addTrainees(selectedEnrollees));
+      handleAddMaterialModal();
+    } catch (error) {
+      console.error('An error occurred while enrolling learners:', error);
```

Comment:
Please use toast message for any failure and also success as well, its better to let the user know about it